.github/workflows/linux.yaml

```diff
@@ -0,0 +1,103 @@
+name: Linux
+
+on: [push, pull_request]
+
+env:
+  REGISTRY_HOST: pcr-internal.puppet.net
+  DOCKER_REGISTRY: pe-and-platform
+  BOLT_SUDO_USER: true
+
+jobs:
+
+  modules:
+    name: Modules
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Setup Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '2.5.x'
+      - name: Install bundler
+        run: |
+          gem install bundler
+          bundle config path vendor/bundle
+      - name: Cache gems
+        id: cache
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile') }}
+      - name: Install gems
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: bundle install --jobs 4 --retry 3
+      - name: Run module tests
+        run: bundle exec rake github:modules
+
+  unit:
+    name: Unit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Setup Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '2.5.x'
+      - name: Install bundler
+        run: |
+          gem install bundler
+          bundle config path vendor/bundle
+      - name: Cache gems
+        id: cache
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile') }}
+      - name: Install gems
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: bundle install --jobs 4 --retry 3
+      - name: Pre-test setup
+        run: |
+          sudo curl -L https://github.com/docker/compose/releases/download/1.23.0/docker-compose-`uname -s`-`uname -m` -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
+          echo 'runner:runner' | sudo chpasswd
+          sudo sh -c "echo 'Defaults authenticate' >> /etc/sudoers"
+          sudo sh -c "echo 'runner  ALL=(ALL) PASSWD:ALL' >> /etc/sudoers"
+          docker-compose -f spec/docker-compose.yml build --parallel ubuntu_node puppet_5_node puppet_6_node
```

Comment:
Do we need to have the docker infrastructure up to run the unit tests? At least locally they work without the docker infra up.

Reply:
Pulled it out. Let's see what happens. :)

Reply:
That appears to be a no: https://github.com/puppetlabs/bolt/commit/9249c5ab652703ae7766fb01c2c2d32808715924/checks?check_suite_id=417505454

Reply:
Ah, ok, I see my confusion. There's a `unit` rake task that *doesn't* require the docker infra. The `travis:unit` (and thus `github:unit`) actually runs all but the most expensive tests, including integration level tests that do rely on the docker infra.

I'm not sure if "cheap" and "expensive" is a better differentiator? At least, that's *why* they're separate jobs. I wonder, since these rake tasks are only used by CI, if it makes sense to just combine them into 'integration tests' and invoke the 'ci' then the 'puppetserver' tasks in order. The reason we do the expensive ones last is so we can fail faster, but I'm not sure we need entirely separate tasks for that?